add wordpress.org active installs shield

koter84 opened the pull request:
i made a new badge to show number of active installs for a wordpress plugin
testing was done on a heroku-app as described in the INSTALL.md

KingYes commented:
+1 cool :)

koter84 commented:
is there information i need to add to get this pull-request merged?

koter84 commented:
@espadrine do you need more information from me for merging this PR? i'd be happy to add or change stuff if needed..

espadrine commented on server.js:
```diff
@@ -4019,6 +4019,44 @@ cache(function(data, match, sendBadge, request) {
+      } else if (total < 10000) {
+        badgeData.colorscheme = 'green';
+      } else {
+        badgeData.colorscheme = 'brightgreen';
```
Could you use `floorCountColor()` to simplify this, please?

espadrine commented on server.js:
```diff
@@ -4019,6 +4019,44 @@ cache(function(data, match, sendBadge, request) {
+  var plugin = match[1];  // eg, `akismet`.
+  var format = match[2];
+  var apiUrl = 'http://api.wordpress.org/plugins/info/1.0/' + plugin + '.json?fields=active_installs';
+  var badgeData = getBadgeData('active installs', data);
```
Do you plan on adding a badge for inactive installs? Would having a badge simply saying "installs" (instead of "active installs") be confusing?

koter84 replied:
wordpress only supplies a 'active installs' value, only active plugins report to wordpress.org that they are in use on a site, so there won't be a inactive installs but i think that showing just installs could be confusing since wordpress.org uses the term 'active installs' as well

espadrine commented on server.js:
```diff
@@ -4019,6 +4019,44 @@ cache(function(data, match, sendBadge, request) {
+      return;
+    }
+    try {
+      var total = JSON.parse(buffer).active_installs + '+';
```
`total` is a string. Do you want to remove the `+ '+'` at the end?

koter84 replied:
wordpress only supplies a ballpark value so no 1325 installs, but 10+ 100+ 1000+ 10K+ etc.
that's why i think there should be a + since this is also how it is displayed on wordpress.org

espadrine commented:
Thanks a lot for your contribution! Sorry about the delay to getting to your PR.

koter84 commented:
i simplified the colors with floorCountColor() and moved the '+' to after the metric()-call which seems better than adding it to total.

koter84 commented:
@espadrine i added some replies to your comments, but because i rebased the code they are not directly visible anymore, so i'll put the remarks here for easier reading.

> Could you use floorCountColor() to simplify this, please?
i changed the code, and now it's using floorCountColor() for determining the colors

> Do you plan on adding a badge for inactive installs? Would having a badge simply saying "installs" (instead of "active installs") be confusing?
there won't be a "inactive installs", but since wordpress uses the term "active installs" i think it's best to use the same terminology

> total is a string. Do you want to remove the + '+' at the end?
i removed the '+' from total, and added it to the badgeData.text[1] which seems more logical. I believe the '+' is needed since wordpress doesn't give a absolute number but 10+, 20+, 30+, 100+, 2000+ etc. so without the '+' it would look like an exact number which it isn't.

please let me know if there is anything which is still unclear, so i can fix or explain it.

koter84 commented:
@espadrine could you please look at this pull-request?
i rebased the code to master, please let me know if you have further questions.
